Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp

```diff
@@ -737,6 +737,19 @@ FloatRect LineLayout::applySVGTextFragments(SVGTextFragmentMap&& fragmentMap)
     return fullBoundaries;
 }
 
+bool LineLayout::preventsSkippingLayout() const
+{
+    // Line layout does not have enough information to tell if layout can be skipped but
+    // it has enough information to tell that it definitely cannot be skipped.
+
+    if (!m_inlineContent)
+        return true;
+
+    return m_inlineContent->displayContent().lines.size() != 1
+        || m_inlineContent->displayContent().boxes.size() != 2
+        || m_inlineContent->displayContent().boxes[0].isText() == m_inlineContent->displayContent().boxes[1].isText();
```

Comment:
isn't m_inlineContent->displayContent().boxes[0].isText() always false since it's the root inline box?

Reply:
I wasn't sure of that, but that's a good point. I can add the `ASSERT` for root and just check `m_inlineContent->displayContent().boxes[1].isText()`